Reject references to references with a better error

References to references currently fail to compile but only because we emit broken C++ code. We should catch this and intentionally report an error explaining that C++ doesn't like references to references.

```rust
mod ffi {
    extern "Rust" {
        type Thing;
        fn repro(r: &&Thing);
    }
}
```

```console
/target/debug/out/repro/src/main.rs.cc:152:44: error: duplicate ‘const’
  152 | void cxxbridge02$repro(const const Thing & &r) noexcept;
      |                              ^~~~~
/target/debug/out/repro/src/main.rs.cc:152:60: error: cannot declare reference to ‘const struct Thing&’, which is not a typedef or a template type argument
  152 | void cxxbridge02$repro(const const Thing & &r) noexcept;
      |                                             ^
```

--- a/syntax/check.rs
+++ b/syntax/check.rs
@@ -230,6 +230,11 @@ fn check_api_fn(cx: &mut Check, efn: &ExternFn) {
                 );
             }
         }
+        if let Type::Ref(ity) = &arg.ty {
+            if let Type::Ref(_) = &ity.inner {
+                cx.error(arg, "Passing a reference to a reference is not supported");
+            }
+        }
     }
 
     if let Some(ty) = &efn.ret {

--- a/tests/ui/reference_to_reference.rs
+++ b/tests/ui/reference_to_reference.rs
@@ -1,0 +1,13 @@
+#[cxx::bridge]
+mod ffi {
+    extern "C" {
+        type ThingC;
+        fn repro_c(t: &&ThingC);
+    }
+    extern "Rust" {
+        type ThingR;
+        fn repro_r(t: &&ThingR);
+    }
+}
+
+fn main() {}

--- a/tests/ui/reference_to_reference.stderr
+++ b/tests/ui/reference_to_reference.stderr
@@ -1,0 +1,11 @@
+error: Passing a reference to a reference is not supported
+ --> $DIR/reference_to_reference.rs:5:20
+  |
+5 |         fn repro_c(t: &&ThingC);
+  |                    ^^^^^^^^^^^
+
+error: Passing a reference to a reference is not supported
+ --> $DIR/reference_to_reference.rs:9:20
+  |
+9 |         fn repro_r(t: &&ThingR);
+  |                    ^^^^^^^^^^^

